daemon/cmd/integration_runner/main.go

```diff
@@ -375,6 +375,7 @@ func main() {
 
 	// Env vars common to all tests.
 	ctx.Env["EXTERNAL_HOST"] = externalHost
+	os.Setenv("EXTERNAL_HOST", externalHost)
```

Comment:
Why is it necessary to add this?

Reply:
In order to use it in `EXPECT_SPAN_EVENTS_LIKE` [here](https://github.com/newrelic/newrelic-php-agent/blob/55ab2076d11456cef2aed9d39f5bd87885679473/tests/integration/external/guzzle7/test_timeout_sync.php#L47).

Reply:
Setting `ctx.Env["EXTERNAL_HOST"]` is not enough for `EXTERNAL_HOST` the env var to be available in the runtime environment for `os.Getenv`, used in [`SubEnvVars`](https://github.com/newrelic/newrelic-php-agent/blob/55ab2076d11456cef2aed9d39f5bd87885679473/daemon/internal/newrelic/integration/test.go#L291), to find it.

Reply:
The code appears to associate the `ctx.Env` with each test run and is used as the env for the PHP which is spawned.

Reply:
It seems currently the `ctx.Env` is not included in the variables the `SubEnvVar` considers.  That might be an enhancement that would remove the need to make this addition.

Reply:
@mfulb 800b45a43ae46043ecdf1f1f0692de7af3fff4ed was needed for `ctx.Env` to be available to env var substitution.

Reply:
@mfulb https://github.com/newrelic/newrelic-php-agent/commit/800b45a43ae46043ecdf1f1f0692de7af3fff4ed wasn't enough - `Test.Env` still didn't have env vars available to PHP. I had to do this: https://github.com/newrelic/newrelic-php-agent/pull/883/commits/e20f4b93b4e7bca34327ab2c8680cf59b84ad9b5. 5e111d2374a53e4383e713cd932e667b0c456fd3 showcases how this works.

Reply:
And possibly the last patch: 5064b7c93267fae793fbef33760b1df37fc35d77, which ensures that values from `/*ENVIRONMENT*/` directive take precedence over OS environment.